### PR TITLE
Protected routes and authentication

### PR DESCRIPTION
**Pull request by flxffy**
This branch needs to be rebased once PR #10 is merged

- Adds login and logout sequences
- Creates components for protected routes - will redirect if user is not logged in
- Updates custom button component to pass unused props to underlying button

### CODE REVIEW

File: src/components/LandingNavbar.jsx [hunk 1]
```diff
@@ -33,7 +37,7 @@ function LandingNavbar() {
         </Navbar.Collapse>
       </div>
       <div className="order-3 ml-auto">
-        <Btn title="Sign in with SSO" type="primary" />
+        <Btn title="Sign in with SSO" type="primary" onClick={() => login(history)} />
```
Comment (hellodommy) [inline src/components/LandingNavbar.jsx:40]: nice

**Review by hellodommy — APPROVED** [review]
Tried on my end and the routing works fine! 👍 
good job

File: src/utils/login.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const login = (history) => {
+  localStorage.setItem("token", "this is a token!");
```
Comment (kenneth-fung) [inline src/utils/login.js:2]: If we use local storage for degree/mod population then the token should probably be named something more specific, but this is alright for now.

**Review by kenneth-fung — APPROVED** [review]
Looks good.

**Pull request merged**